sabnzbd/sorting.py

```diff
@@ -444,350 +297,236 @@ def construct_path(self):
         # Lowercase all characters wrapped in {}
         path = to_lowercase(path)
 
-        # Strip any extra ' ' '.' or '_' around foldernames
-        path = strip_folders(path)
+        # Strip any extra spaces, dots, and underscores around directory names
+        path = strip_path_elements(path)
```

Comment:
A comment in the code said windows cannot handle path elements ending in a dot; I assumed that's (still) accurate. But even ignoring that, stripping the path elements is useful in case certain %patterns are replaced with nothing, resulting in ugly dangling separators. The function was broken before, in the sense that the success of the cleanup routine depended on the order of appearance of the chars it should strip.